streaming/src/main/scala/org/apache/gearpump/streaming/dsl/Stream.scala

```diff
@@ -147,6 +150,33 @@ class Stream[T](
     graph.addEdge(thisNode, edge.getOrElse(Shuffle), processorOp)
     new Stream[R](graph, processorOp, Some(Shuffle))
   }
+
+  private def groupByWindow[GROUP](fn: GroupByFn[T, GROUP],
+      parallelism: Int, description: String): GroupByStream[T, GROUP] = {
+    val groupOp = GroupByOp[T, GROUP](fn,
+      parallelism, description)
+    graph.addVertex(groupOp)
+    graph.addEdge(thisNode, edge.getOrElse(Shuffle), groupOp)
+    new GroupByStream[T, GROUP](graph, groupOp)
+  }
+
+}
+
+class GroupByStream[T, GROUP](
+    private val graph: Graph[Op, OpEdge],
+    private val groupByOp: GroupByOp[T, GROUP],
+    private val edge: Option[OpEdge] = None) extends Stream[T](graph, groupByOp, edge) {
```

Comment:
Is 'private val graph' equivalent to 'graph' (val adds public accessors so private takes them away?)